core/src/main/java/io/questdb/cutlass/http/processors/LineHttpProcessor.java

```diff
@@ -0,0 +1,203 @@
+            return;
+        }
+
+        byte timestampPrecision;
+        DirectUtf8Sequence precision = requestHeader.getUrlParam(URL_PARAM_PRECISION);
+        if (precision != null) {
+            int len = precision.size();
+            if ((len == 1 && precision.byteAt(0) == 'n') || (len == 2 && precision.byteAt(0) == 'n' && precision.byteAt(1) == 's')) {
+                // V2 influx client sends "n" and V3 sends "ns"
+                timestampPrecision = ENTITY_UNIT_NANO;
+            } else if ((len == 1 && precision.byteAt(0) == 'u') || (len == 2 && precision.byteAt(0) == 'u' && precision.byteAt(1) == 's')) {
+                // V2 influx client sends "u" and V3 sends "us"
+                timestampPrecision = ENTITY_UNIT_MICRO;
+            } else if (len == 2 && precision.byteAt(0) == 'm' && precision.byteAt(1) == 's') {
+                timestampPrecision = ENTITY_UNIT_MILLI;
+            } else if (len == 1 && precision.byteAt(0) == 's') {
+                timestampPrecision = ENTITY_UNIT_SECOND;
+            } else if (len == 1 && precision.byteAt(0) == 'm') {
+                timestampPrecision = ENTITY_UNIT_MINUTE;
+            } else if (len == 1 && precision.byteAt(0) == 'h') {
+                timestampPrecision = ENTITY_UNIT_HOUR;
+            } else {
+                LOG.info().$("unsupported precision [url=")
+                        .$(requestHeader.getUrl())
+                        .$(", precision=").$(precision)
+                        .I$();
+                state.reject(PRECISION_NOT_SUPPORTED, "unsupported precision");
+                return;
+            }
+        } else {
+            timestampPrecision = ENTITY_UNIT_NANO;
+        }
+
+        state.of(context.getFd(), timestampPrecision, context.getSecurityContext());
+    }
+
+    @Override
+    public void onPartBegin(HttpRequestHeader partHeader) {
+    }
+
+    @Override
+    public void onPartEnd() {
+    }
+
+    @Override
+    public void onRequestComplete(HttpConnectionContext context) throws PeerDisconnectedException, PeerIsSlowToReadException {
+        state.onMessageComplete();
+        if (state.isOk()) {
+            state.commit();
+        }
+        // Check state again, commit may have failed
+        if (state.isOk()) {
+            state.setSendStatus(SendStatus.HEADER);
+            context.simpleResponse().sendStatus(204);
+        } else {
+            state.setSendStatus(SendStatus.HEADER);
+            sendErrorHeader(context);
+            state.setSendStatus(SendStatus.CONTENT);
+            sendErrorContent(context);
+        }
+    }
+
+    @Override
+    public void resumeRecv(HttpConnectionContext context) {
+        state = LV.get(context);
+    }
+
+    @Override
+    public void resumeSend(
+            HttpConnectionContext context
+    ) throws PeerDisconnectedException, PeerIsSlowToReadException, ServerDisconnectException, QueryPausedException {
+        state = LV.get(context);
+        if (state == null) {
```

Comment:
is this possible at all? 🤔

Reply:
Unlikely, I can replace it with an assert

Reply:
that would be nice - it would help to understand the expected lifecycle. our HTTP state machine is already pretty hard to grasp as it is.

Reply:
done in [1380437](https://github.com/questdb/questdb/pull/4034/commits/13804375a595e586fde3ef559ac83e92e9da2e1b)